nemo/collections/diffusion/models/flux/model.py

```diff
@@ -462,6 +463,11 @@ def __init__(
         self.model_type = ModelType.encoder_or_decoder
         self.text_precached = self.t5_params is None or self.clip_params is None
         self.image_precached = self.vae_config is None
+        self.seed = seed
+
+    def setup(self, stage: str):
+        super().setup(stage)
+        torch.manual_seed(self.seed + 100 * parallel_state.get_data_parallel_rank())
```

Comment:
IIUC, I think we should use global rank instead of data parallel rank, because if we use other parallelisms then there's still the chance two ranks have the same seed.

Reply:
Hmm, I was thinking the noise map part for diffusion model, but I just realized that noise map should be deemed as data, so the ones with the same data rank should have the same seed.